test/common/run-tests

```diff
@@ -183,12 +183,8 @@ class Test:
             return
 
         print()  # Tap needs to start on a separate line
-        if self.returncode == 0:
-            print(f"ok {self}{retry_reason}")
-        elif skip_reason:
-            print(f"ok {self}{skip_reason}")
-        else:
-            print(f"not ok {self}{retry_reason}")
+        status = 'ok' if self.returncode in [0, 77] else 'not ok'
```

Comment:
It's this change which breaks the "SKIP" case. The returncode is 1 in the failed case, because finish() calls `test-failure-policy` without adjusting the test's `self.returncode`. I added a fixup.

Reply:
Pushed.